src/main/scala/mesosphere/marathon/core/election/impl/CuratorElectionService.scala

```diff
@@ -67,13 +68,54 @@ class CuratorElectionService(
     maybeLatch.get.start()
   }
 
+  override protected def startLeadership(abdicate: Abdicator): Unit = {
+    maybeLatch match {
+      case None => super.startLeadership(abdicate)
+      case Some(l) =>
+        l.synchronized {
+          super.startLeadership(abdicate)
+        }
+    }
+  }
+
+  override protected def stopLeadership(): Unit = {
+    maybeLatch match {
+      case None => super.stopLeadership
+      case Some(l) =>
+        l.synchronized {
```

Comment:
Hm, I'm fine with this fix but wonder if we could avoid the `synchronized` and be a little more explicit.